components/dashboard/src/components/create/create-workspace.tsx

```diff
@@ -299,7 +321,7 @@ export class CreateWorkspace extends React.Component<CreateWorkspaceProps, Creat
                         service={this.props.service}
                         prebuildingWorkspaceId={runningPrebuild.workspaceID}
                         justStarting={runningPrebuild.starting}
-                        onBuildDone={() => this.doCreateWorkspaceWithMode(CreateWorkspaceMode.UsePrebuild)}
+                        onWatchPrebuild={() => this.doCreateWorkspaceWithMode(CreateWorkspaceMode.UsePrebuild)}
```

Comment:
still thinking, could `this.doCreateWorkspaceWithMode(CreateWorkspaceMode.UsePrebuild)` be called twice, and does it matter?

Reply:
I'm looking how to simplify it. The code generally can get in race conditions because of setInterval and disconnections, i.e. multiple calls to the server each 10 seconds which are hanging and then fail all together. It makes reconnection to look very slow. We should do better.